Bug: 小红书解析失败

### 描述问题

小红书解析失败


### 复现步骤

链接： https://www.xiaohongshu.com/explore/691e68a8000000001e02bcda?app_platform=android&ignoreEngage=true&app_version=9.12.0&share_from_user_hidden=true&xsec_source=app_share&type=normal&xsec_token=CBunzr4Cq8N7jbcXqpWDxGn11k7XwVIJ59KOvkRS_Qabw=&author_share=1&xhsshare=&shareRedId=N0w7NTk7ND82NzUyOTgwNjY0OTc4Sz9N&apptime=1767276999&share_id=4671bb47e5e64f0db4663fbd23de205d&share_channel=qq

原消息是个卡片，这里是浏览器打开的链接

### 插件版本

2.3.6

### 期望的结果

_No response_

### 截图或日志

```log
01-01 23:03:12 [ERROR] nonebot | Running Matcher(type='message', module=nonebot_plugins.nonebot_plugin_parser.matchers, lineno=52) failed.
Traceback (most recent call last):

  File "E:\Bot\bot.py", line 30, in <module>
    nonebot.run()
    │       └ <function run at 0x000001A0EE4C28E0>
    └ <module 'nonebot' from 'C:\\Users\\Administrator\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\zhenxun-bot-oFP0POfx-py3.11\\...

  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\__init__.py", line 337, in run
    get_driver().run(*args, **kwargs)
    │                 │       └ {}
    │                 └ ()
    └ <function get_driver at 0x000001A0EB7E45E0>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
    │       └ <function run at 0x000001A0EF4E1080>
    └ <module 'uvicorn' from 'C:\\Users\\Administrator\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\zhenxun-bot-oFP0POfx-py3.11\\...
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\uvicorn\main.py", line 593, in run
    server.run()
    │      └ <function Server.run at 0x000001A0EF4E1260>
    └ <uvicorn.server.Server object at 0x000001A084BC0650>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\uvicorn\server.py", line 67, in run
    return asyncio_run(self.serve(sockets=sockets), loop_factory=self.config.get_loop_factory())
           │           │    │             │                      │    │      └ <function Config.get_loop_factory at 0x000001A0EF4E0D60>
           │           │    │             │                      │    └ <uvicorn.config.Config object at 0x000001A086216050>
           │           │    │             │                      └ <uvicorn.server.Server object at 0x000001A084BC0650>
           │           │    │             └ None
           │           │    └ <function Server.serve at 0x000001A0EF4E1300>
           │           └ <uvicorn.server.Server object at 0x000001A084BC0650>
           └ <function asyncio_run at 0x000001A0EF49D6C0>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\uvicorn\_compat.py", line 30, in asyncio_run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x000001A0865B1B70>
           │      └ <function Runner.run at 0x000001A0EDCE5D00>
           └ <asyncio.runners.Runner object at 0x000001A0861B6810>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\z...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x000001A0EDCDF880>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x000001A0861B6810>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 641, in run_until_complete
    self.run_forever()
    │    └ <function ProactorEventLoop.run_forever at 0x000001A0EDD54CC0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 608, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x000001A0EDCE5620>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1936, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x000001A0EC4A9F80>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\utils.py", line 259, in run_coro_with_shield
    return await coro
                 └ <coroutine object check_and_run_matcher at 0x000001A0894C9EA0>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\message.py", line 508, in check_and_run_matcher
    await _run_matcher(
          └ <function _run_matcher at 0x000001A0EEE12840>
> File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\message.py", line 460, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
          │       │   │    │      │      │      └ {<function _uni_msg at 0x000001A0F0C03920>: <nonebot.internal.params.DependencyCache object at 0x000001A086AC5310>, <function...
          │       │   │    │      │      └ <contextlib.AsyncExitStack object at 0x000001A086C07850>
          │       │   │    │      └ {'_prefix': {'command': None, 'raw_command': None, 'command_arg': None, 'command_start': None, 'command_whitespace': None}, '...
          │       │   │    └ GroupMessageEvent(time=1767279790, self_id=000, post_type='message', sub_type='normal', user_id=000, message_typ...
          │       │   └ Bot(type='OneBot V11', self_id='000')
          │       └ <function Matcher.run at 0x000001A0EE4FA020>
          └ Matcher(type='message', module=nonebot_plugins.nonebot_plugin_parser.matchers, lineno=52)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 926, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
          │    │          │    │      │      │      └ {<function _uni_msg at 0x000001A0F0C03920>: <nonebot.internal.params.DependencyCache object at 0x000001A086AC5310>, <function...
          │    │          │    │      │      └ <contextlib.AsyncExitStack object at 0x000001A086C07850>
          │    │          │    │      └ {'_prefix': {'command': None, 'raw_command': None, 'command_arg': None, 'command_start': None, 'command_whitespace': None}, '...
          │    │          │    └ GroupMessageEvent(time=1767279790, self_id=000, post_type='message', sub_type='normal', user_id=000, message_typ...
          │    │          └ Bot(type='OneBot V11', self_id='000')
          │    └ <function Matcher.simple_run at 0x000001A0EE4F9F80>
          └ Matcher(type='message', module=nonebot_plugins.nonebot_plugin_parser.matchers, lineno=52)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 863, in simple_run
    await handler(
          └ Dependent(call=parser_handler)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\dependencies\__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
                 │    │             │         │    │    │       └ {'sr': <nonebot_plugins.nonebot_plugin_parser.matchers.rule.SearchResult object at 0x000001A08948DE00>}
                 │    │             │         │    │    └ <function parser_handler at 0x000001A0850D3F60>
                 │    │             │         │    └ Dependent(call=parser_handler)
                 │    │             │         └ ~R
                 │    │             └ <class 'collections.abc.Awaitable'>
                 │    └ typing.Callable
                 └ <function cast at 0x000001A0EBE33600>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_parser\helper.py", line 193, in wrapper
    result = await func(*args, **kwargs)
                   │     │       └ {'sr': <nonebot_plugins.nonebot_plugin_parser.matchers.rule.SearchResult object at 0x000001A08948DE00>}
                   │     └ ()
                   └ <function parser_handler at 0x000001A0850D3EC0>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_parser\matchers\__init__.py", line 76, in parser_handler
    result = await parser.parse(sr.keyword, sr.searched)
                   │      │     │  │        │  └ <member 'searched' of 'SearchResult' objects>
                   │      │     │  │        └ <nonebot_plugins.nonebot_plugin_parser.matchers.rule.SearchResult object at 0x000001A08948DE00>
                   │      │     │  └ <member 'keyword' of 'SearchResult' objects>
                   │      │     └ <nonebot_plugins.nonebot_plugin_parser.matchers.rule.SearchResult object at 0x000001A08948DE00>
                   │      └ <function BaseParser.parse at 0x000001A084C66A20>
                   └ <nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.XiaoHongShuParser object at 0x000001A0881AF590>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_parser\parsers\base.py", line 110, in parse
    return await self._handlers[keyword](self, searched)
                 │    │         │        │     └ <re.Match object; span=(28, 394), match='discovery/item/691e68a8000000001e02bcda?app_platf>
                 │    │         │        └ <nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.XiaoHongShuParser object at 0x000001A0881AF590>
                 │    │         └ 'shu.com/discovery'
                 │    └ {'shu.com/discovery': <function XiaoHongShuParser._parse_discovery at 0x000001A084C7AAC0>, 'shu.com/explore': <function XiaoH...
                 └ <nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.XiaoHongShuParser object at 0x000001A0881AF590>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_parser\parsers\xiaohongshu\__init__.py", line 59, in _parse_discovery
    return await self.parse_explore(f"https://www.xiaohongshu.com/{explore_route}", xhs_id)
                 │    │                                                             └ '691e68a8000000001e02bcda'
                 │    └ <function XiaoHongShuParser.parse_explore at 0x000001A084C7A8E0>
                 └ <nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.XiaoHongShuParser object at 0x000001A0881AF590>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_parser\parsers\xiaohongshu\__init__.py", line 77, in parse_explore
    init_state = explore.decoder.decode(raw)
                 │       │       │      └ '{"global":{"appSettings":{"notificationInterval":30,"prefetchTimeout":3001,"prefetchRedisExpires":259200000,"searchFilterGui...
                 │       │       └ <method 'decode' of 'msgspec.json.Decoder' objects>
                 │       └ msgspec.json.Decoder(<class 'nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.explore.InitialState'>)
                 └ <module 'nonebot_plugins.nonebot_plugin_parser.parsers.xiaohongshu.explore' from 'E:\\Bot\\nonebot_plugins\\nonebot_plugin_pa...

msgspec.ValidationError: Object missing required field `type` - at `$.note.noteDetailMap[...].note`
```

--- a/src/nonebot_plugin_parser/parsers/xiaohongshu/__init__.py
+++ b/src/nonebot_plugin_parser/parsers/xiaohongshu/__init__.py
@@ -30,32 +30,23 @@ class XiaoHongShuParser(BaseParser):
         }
         self.ios_headers.update(discovery_headers)
 
-    @handle("xhslink.com", r"xhslink\.com/[A-Za-z0-9._?%&+=/#@-]*")
+    @handle("xhslink.com", r"xhslink\.com/[A-Za-z0-9._?%&+=/#@-]+")
     async def _parse_short_link(self, searched: re.Match[str]):
         url = f"https://{searched.group(0)}"
         return await self.parse_with_redirect(url, self.ios_headers)
 
     # https://www.xiaohongshu.com/explore/68feefe40000000007030c4a?xsec_token=ABjAKjfMHJ7ck4UjPlugzVqMb35utHMRe_vrgGJ2AwJnc=&xsec_source=pc_feed
-    @handle(
-        "shu.com/explore",
-        r"explore/(?P<query>(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*)",
-    )
     # https://www.xiaohongshu.com/discovery/item/68e8e3fa00000000030342ec?app_platform=android&ignoreEngage=true&app_version=9.6.0&share_from_user_hidden=true&xsec_source=app_share&type=normal&xsec_token=CBW9rwIV2qhcCD-JsQAOSHd2tTW9jXAtzqlgVXp6c52Sw%3D&author_share=1&xhsshare=QQ&shareRedId=ODs3RUk5ND42NzUyOTgwNjY3OTo8S0tK&apptime=1761372823&share_id=3b61945239ac403db86bea84a4f15124&share_channel=qq
-    @handle(
-        "shu.com/discovery",
-        r"discovery/item/(?P<query>(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*)",
-    )
+    @handle("xiaohongshu.com", r"(explore|discovery/item)/(?P<query>(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]+)")
     async def _parse_common(self, searched: re.Match[str]):
-        query, xhs_id = searched.group("query", "xhs_id")
         xhs_domain = "https://www.xiaohongshu.com"
-        explore_query = f"{xhs_domain}/explore/{query}"
+        query, xhs_id = searched.group("query", "xhs_id")
 
         try:
-            return await self.parse_explore(explore_query, xhs_id)
+            return await self.parse_explore(f"{xhs_domain}/explore/{query}", xhs_id)
         except Exception as e:
             logger.warning(f"parse_explore failed, error: {e}, fallback to parse_discovery")
-            discovery_query = f"{xhs_domain}/discovery/item/{query}"
-            return await self.parse_discovery(discovery_query)
+            return await self.parse_discovery(f"{xhs_domain}/discovery/item/{query}")
 
     async def parse_explore(self, url: str, xhs_id: str):
         from . import explore

--- a/src/nonebot_plugin_parser/parsers/xiaohongshu/__init__.py
+++ b/src/nonebot_plugin_parser/parsers/xiaohongshu/__init__.py
@@ -38,28 +38,24 @@ class XiaoHongShuParser(BaseParser):
     # https://www.xiaohongshu.com/explore/68feefe40000000007030c4a?xsec_token=ABjAKjfMHJ7ck4UjPlugzVqMb35utHMRe_vrgGJ2AwJnc=&xsec_source=pc_feed
     @handle(
         "shu.com/explore",
-        r"explore/(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*",
+        r"explore/(?P<query>(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*)",
     )
-    async def _parse_explore(self, searched: re.Match[str]):
-        url = f"https://www.xiaohongshu.com/{searched.group(0)}"
-        xhs_id = searched.group("xhs_id")
-        return await self.parse_explore(url, xhs_id)
-
     # https://www.xiaohongshu.com/discovery/item/68e8e3fa00000000030342ec?app_platform=android&ignoreEngage=true&app_version=9.6.0&share_from_user_hidden=true&xsec_source=app_share&type=normal&xsec_token=CBW9rwIV2qhcCD-JsQAOSHd2tTW9jXAtzqlgVXp6c52Sw%3D&author_share=1&xhsshare=QQ&shareRedId=ODs3RUk5ND42NzUyOTgwNjY3OTo8S0tK&apptime=1761372823&share_id=3b61945239ac403db86bea84a4f15124&share_channel=qq
     @handle(
         "shu.com/discovery",
-        r"discovery/item/(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*",
+        r"discovery/item/(?P<query>(?P<xhs_id>[0-9a-zA-Z]+)\?[A-Za-z0-9._%&+=/#@-]*)",
     )
-    async def _parse_discovery(self, searched: re.Match[str]):
-        route = searched.group(0)
-        explore_route = route.replace("discovery/item", "explore", 1)
-        xhs_id = searched.group("xhs_id")
+    async def _parse_common(self, searched: re.Match[str]):
+        query, xhs_id = searched.group("query", "xhs_id")
+        xhs_domain = "https://www.xiaohongshu.com"
+        explore_query = f"{xhs_domain}/explore/{query}"
 
         try:
-            return await self.parse_explore(f"https://www.xiaohongshu.com/{explore_route}", xhs_id)
-        except ParseException:
-            logger.debug("parse_explore failed, fallback to parse_discovery")
-            return await self.parse_discovery(f"https://www.xiaohongshu.com/{route}")
+            return await self.parse_explore(explore_query, xhs_id)
+        except Exception as e:
+            logger.warning(f"parse_explore failed, error: {e}, fallback to parse_discovery")
+            discovery_query = f"{xhs_domain}/discovery/item/{query}"
+            return await self.parse_discovery(discovery_query)
 
     async def parse_explore(self, url: str, xhs_id: str):
         from . import explore

--- a/tests/parsers/test_xhs.py
+++ b/tests/parsers/test_xhs.py
@@ -12,7 +12,9 @@ async def test_parse():
 
     parser = XiaoHongShuParser()
     urls = [
-        "https://www.xiaohongshu.com/explore/68f963260000000004005843?xsec_token=ABn8T0lrJdPfRlWpNvLoYWTd0x54P44tsJtYgqhujHTjo=&xsec_source=pc_feed"
+        "https://www.xiaohongshu.com/explore/68f963260000000004005843?xsec_token=ABn8T0lrJdPfRlWpNvLoYWTd0x54P44tsJtYgqhujHTjo=&xsec_source=pc_feed",
+        "https://www.xiaohongshu.com/explore/691e68a8000000001e02bcda?app_platform=android&ignoreEngage=true&app_version=9.12.0&share_from_user_hidden=true&xsec_source=app_share&type=normal&xsec_token=CBunzr4Cq8N7jbcXqpWDxGn11k7XwVIJ59KOvkRS_Qabw=&author_share=1&xhsshare=&shareRedId=N0w7NTk7ND82NzUyOTgwNjY0OTc4Sz9N&apptime=1767276999&share_id=4671bb47e5e64f0db4663fbd23de205d&share_channel=qq",
+        "https://www.xiaohongshu.com/discovery/item/691e68a8000000001e02bcda?source=webshare&xhsshare=pc_web&xsec_token=CBunzr4Cq8N7jbcXqpWDxGn11k7XwVIJ59KOvkRS_Qabw=&xsec_source=pc_share",
     ]
 
     async def parse(url: str) -> None:

